Agoda.Frameworks.Http.Tests/RandomUrlHttpClientTest.cs

```diff
@@ -156,16 +158,91 @@ public async Task TestHttpRequestException()
 
             Assert.IsAssignableFrom<ServiceUnavailableException>(result.Last().Exception);
             Assert.AreEqual(3, result.Count);
-        }
-
-        [Test]
-        public void UpdateBaseUrls()
-        {
+        }
+
+        [Test]
+        public void UpdateBaseUrls()
+        {
             var client = new RandomUrlHttpClient(new[] { "http://test/1", "http://test/2" });
 
             // Should not throw due to identical items in the list (Distinct before ToDictionary)
             client.UpdateBaseUrls(new[] { "http://test/3", "http://test/3" });
-            Assert.AreEqual("http://test/3", client.UrlResourceManager.SelectRandomly());
-        }
-    }
+            Assert.AreEqual("http://test/3", client.UrlResourceManager.SelectRandomly());
+        }
+
+        [Test]
+        public void ShouldThrowRequestTimeoutExceptionWhenHttpRequestTimesOut()
+        {
+            var mockHttp = new WaitedMockHttpMessageHandler(new TimeSpan(0, 0, 3));
+
+            mockHttp.When(HttpMethod.Get, "http://test/*")
+                .Respond(msg =>
+                {
+                    StringAssert.IsMatch("http://test/1|2/api/55", msg.RequestUri.AbsoluteUri);
+                    return new StringContent("ok");
+                });
+            var httpclient = mockHttp.ToHttpClient();
+
+            var client = new RandomUrlHttpClient(
+                httpclient,
+                new[] { "http://test/1", "http://test/2" },
+                new TimeSpan(0, 0, 1),

```

Comment:
Might be a bit nit-picky, but is there any reason not to bring this down an order of magnitude (to milliseconds) to speed up unittests?

Even more nit-picky, so feel free to disagree, but I find the `TimeSpan.FromSeconds()` / `TimeSpan.FromMilliseconds()` syntax to be way more intuitive than having to count parameters in the constructor signature in a PR :)

Reply:
I updated the PR according to the comment